### PR TITLE
UIF-34 Add name and code validation on Fund form

### PR DESCRIPTION
**Pull request by aliaksei-chumakou**
also switch to the final form.
Since some "warning" validation for name was added, I've decided we need to implement this on the new api of reac-final-form, because it's a default component for Forms in folio. 

## Purpose
Do not allow the duplication of Fund codes.
https://issues.folio.org/browse/UIF-34

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -129,17 +132,23 @@
     "eslint-plugin-filenames": "^1.3.2",
     "eslint-plugin-no-only-tests": "^2.3.1",
     "eslint-plugin-react-hooks": "^2.0.1",
-    "react-intl": "^2.4.0",
     "react": "^16.5.1",
     "react-dom": "^16.5.1",
+    "react-intl": "^2.4.0",
```
Comment (NikitaSedyx) [inline package.json]: react + intl dev depth? hm
not related to PR changes

File: src/components/Fund/FundForm/FundForm.js [hunk 2]
```diff
@@ -64,45 +64,65 @@ const getLastMenu = (handleSubmit, pristine, submitting) => {
...
   const funds = sortBy(get(parentResources, ['records', 'records'], []), 'name');
   const fundTypes = get(parentResources, ['fundTypes', 'records'], []).map(
     ({ name, id }) => ({
       label: name,
       value: id,
     }),
   );
+
+  const validateFundName = async value => {
+    const isRequired = validateRequired(value);
```
Comment (NikitaSedyx) [inline src/components/Fund/FundForm/FundForm.js]: if it' an error it's better to name `errorRequired` or smth like

File: package.json (hunk 1 above)
Comment (aliaksei-chumakou) [reply package.json]: It looks like I've added new packages and they were re-arranged in alphabetical order

Comment (NikitaSedyx) [reply package.json]: yep, but why is react in devDepth?  better to ask Arvind I suppose...

Comment (aliaksei-chumakou) [reply package.json]: I've moved it to deps

**Comment by aliaksei-chumakou** [comment]
Tests coverage is below 80% since there is no easy way to emulate save and validation things for react-final-form

**Pull request merged**